File: templates/algorithums/primeQuestions.cpp
vector<int> isprime, prime, SPF;
void manipulated_seive(int MAXN) {
  isprime.assign(MAXN, 1), SPF.resize(MAXN);
  isprime[0] = isprime[1] = 0;
  for (int i = 2; i < MAXN; i++) {
    if (isprime[i])
      prime.push_back(i), SPF[i] = i;
    for (int j = 0; j < prime.size() && i * prime[j] < MAXN && prime[j] <= SPF[i]; j++) {
      isprime[i * prime[j]] = false;
      SPF[i * prime[j]] = prime[j];
    }
  }
}

File: python/templates/algorithums/primeQuestions.py
isprime, prime, SPF = [], [], []

def manipulated_seive(MAXN : int) -> None:
  global isprime, prime, SPF
  isprime, SPF = [1] * (MAXN + 1), [1] * (MAXN + 1)
  isprime[0], isprime[1] = 0, 0
  for i in range(2, MAXN):
    if isprime[i]:
      prime.append(i)
      SPF[i] = i
    j = 0
    while j < len(prime) and i * prime[j] < MAXN and prime[j] <= SPF[i]:
      isprime[i * prime[j]] = 0
      SPF[i * prime[j]] = prime[j]
      j += 1


manipulated_seive(100)

print(SPF)